logstash-core/spec/logstash/shutdown_watcher_spec.rb

```diff
@@ -69,8 +59,9 @@
       it "shouldn't force the shutdown" do
         expect(subject).to_not receive(:force_exit)
         thread = Thread.new(subject) {|subject| subject.start }
-        sleep 0.1 until report_count > check_threshold
-        thread.kill
+        sleep 0.1 until subject.attempts_count > check_threshold
+        subject.stop!
+        thread.join
```

Comment:
@ph we may wanna join with a timeout here right, given how much fun a thread dump is from JRuby right and assert that it worked?